### PR TITLE
SDDM theme with Autologin (My very first PR)

### PR DESCRIPTION
**Pull request by mashrukk**
# Pull Request

## SDDM Astronaut theme with Autologin
This script installs sddm-astronaut-theme from last stream and provides an option to enable autologin with sddm. Users can choose which desktop session they want to autologin on startup. After selecting desktop session, it will do a sddm restart and autologin to that desktop.

## Testing
If you want to test it first, do the following.
```
git clone --branch sddm https://github.com/mashrukk/linutil
cd linutil && cargo run
```

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] UI/UX improvement

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: tabs/applications-setup/sddm-theme.sh [hunk 1]
```diff
@@ -0,0 +1,115 @@
+#!/bin/bash -e
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:1]:
```suggestion
#!/bin/sh -e
```

File: tabs/applications-setup/sddm-theme.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+EOF
+
+    $ESCALATION_TOOL systemctl enable sddm
+    echo "SDDM theme configuration complete."
+}
+
+# Autologin
+configure_autologin() {
+    echo "Available sessions:"
+    declare -a session_names
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:62]: This is not posix compliant.

File: tabs/applications-setup/sddm-theme.sh [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    declare -A sessions
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:63]: This is not posix compliant.

File: tabs/applications-setup/sddm-theme.sh [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    i=1
+    for session_type in xsessions wayland-sessions; do
+        for session_file in /usr/share/$session_type/*.desktop; do
+            [ -e "$session_file" ] || continue
+            name=$(grep -i "^Name=" "$session_file" | cut -d= -f2)
+            type=${session_type%s}  # Remove trailing 's'
+            sessions["$i"]="$session_file"
+            session_names[i]="$name ($type)"
+            echo "$i) $name ($type)"
+            ((i++))
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:74]: This is not posix compliant.

File: tabs/applications-setup/sddm-theme.sh [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        done
+    done
+
+    # Prompt user to choose a session
+    while true; do
+        read -p "Enter the number of the session you'd like to autologin: " choice
+        if [[ -n "${sessions[$choice]}" ]]; then
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:81]: This is not posix compliant.

File: tabs/applications-setup/sddm-theme.sh [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+            session_file="${sessions[$choice]}"
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:82]: This is not posix compliant.

File: tabs/applications-setup/sddm-theme.sh [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+if ! command -v sddm &> /dev/null; then
+    install_sddm
+else
+    echo "SDDM is already installed. Skipping installation."
+fi
+
+install_theme
+
+read -p "Do you want to enable autologin? (y/n): " enable_autologin
+if [[ $enable_autologin =~ ^[Yy]$ ]]; then
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:109]: This is not posix compliant.

File: tabs/applications-setup/sddm-theme.sh [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+    # Find the corresponding .desktop file and Update SDDM configuration
+    actual_session=$(basename "$session_file" .desktop)
+
+    $ESCALATION_TOOL sed -i '1i[Autologin]\nUser = '$USERNAME'\nSession = '$actual_session'\n' /etc/sddm.conf
+    echo "Autologin configuration complete."
+}
+
+checkEnv
+checkEscalationTool
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:97]:
```
checkEnv
checkEscalationTool
```
These need to be at the bottom of the script

**Comment by mashrukk** [comment]
@nnyyxxxx i'm redoing the PR making it posix compliant.

**Comment by nnyyxxxx** [comment]
> @nnyyxxxx i'm redoing the PR making it posix compliant.

Thanks.

**Pull request closed without merging**